### PR TITLE
mlflow - connecting endpoints to tracking server

### PR DESCRIPTION
**Pull request by tayyib-saddique**
## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [ ] I have tested any code added or changed
- [ ] I have run integration tests

### CODE REVIEW

File: mlflow/docker/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
 FROM ghcr.io/mlflow/mlflow
 
 WORKDIR /mlflow
 
-RUN apt-get update && apt-get install -y \
-    curl \
-    && rm -rf /var/lib/apt/lists/*
+COPY mlflow/docker/requirements.txt .
+RUN pip install --upgrade pip && \
+    pip install -r requirements.txt
```
[the remark this answers is not in the document]
Comment (tayyib-saddique) [reply mlflow/docker/Dockerfile]: Unfortunately poetry does not seem to support proper installation of MLflow. Have tested and MLflow fails to be properly installed or fails to connect the tracking server with LLM endpoints. 
https://github.com/mlflow/mlflow/issues/9717

Have reverted back to pip with a requirements.txt file for simplicity based on steer.

**Comment by tayyib-saddique** [comment]
> Where is the tracking server storing experiments? Is the intention to deploy this so we can all see each other's experiments, and will this setup work that way?

**Comment by tayyib-saddique** [comment]
> Where is the tracking server storing experiments? Is the intention to deploy this so we can all see each other's experiments, and will this setup work that way?

Tracking server stores experiments locally - have discussed potential possibility to deploy in AWS so experiments can be shared in line with broader work related to DeepEval and Ragas

File: mlflow/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FROM ghcr.io/mlflow/mlflow
```
Comment (andy-symonds) [inline mlflow/Dockerfile:1]: Should we pin image version? e.g. ghcr.io/mlflow/mlflow:v2.12.1

File: mlflow/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+pydantic
+starlette
+limits
+watchfiles
+fastapi
+slowapi
+uvicorn
+aiohttp
+gunicorn
+boto3
```
Comment (andy-symonds) [inline mlflow/requirements.txt:11]: Can we pin versions to each package? Particularly important with this being the first non-poetry managed service.

File: mlflow/Dockerfile (hunk 2 above)
Comment (tayyib-saddique) [reply mlflow/Dockerfile]: Pinned to v2:12.1

File: mlflow/requirements.txt (hunk 3 above)
Comment (tayyib-saddique) [reply mlflow/requirements.txt]: Pinned versions to each package

File: mlflow/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+mlflow
```
Comment (andy-symonds) [inline mlflow/requirements.txt]: Is `mlflow` along okay? Or do we need `mlflow[genai]`? See https://mlflow.org/docs/latest/llms/deployments/index.html

File: mlflow/config.yaml
Comment (andy-symonds) [inline mlflow/config.yaml:1]: Do we have/need a ticket in the backlog to explore how to switch these OpenAI endpoints to our Redbox CoreAPI endpoint?

File: docker-compose.yml [hunk 5]
```diff
@@ -33,14 +33,17 @@ services:
     volumes:
       - ./:/app/:z
```
[the remark this answers is not in the document]
Comment (tayyib-saddique) [reply docker-compose.yml:34]: Outside scope of PR - the volume is in reference to`django-app`, not `mlflow`

File: mlflow/config.yaml
Comment (tayyib-saddique) [reply mlflow/config.yaml:1]: Perhaps one to include in the backlog - definitely a good shout to explore connecting to the Redbox CoreAPI endpoint

**Comment by andy-symonds** [comment]
> Tracking server stores experiments locally - have discussed potential possibility to deploy in AWS so experiments can be shared in line with broader work related to DeepEval and Ragas
Deploying to AWS is a separate ticket/piece of work, so let's add that to the backlog to track it.

**Review by andy-symonds — APPROVED** [review]

**Pull request merged**